packages/minter-contracts/test-hs/Test/Marketplace/FA2FixedFee.hs

```diff
@@ -0,0 +1,386 @@
+      contract <- originateMarketplaceContract setup
+      consumer <- originateSimple "balance-response-consumer" [] (contractConsumer @[FA2.BalanceResponseItem])
+
+      withSender seller $
+        sell testData setup contract
+      withSender buyer $
+        buy setup contract
+
+      let expectedFee = ceiling @Double @Natural (fromIntegral (testSalePrice * testFeePercent) / 100)
+      feeCollectorsBalance <- balanceOf moneyFA2 consumer moneyTokenId feeCollector
+      feeCollectorsBalance @== expectedFee
+
+hprop_Tokens_are_transferred_to_seller :: Property
+hprop_Tokens_are_transferred_to_seller  =
+  property $ do
+    testData@TestData{testSalePrice, testFeePercent} <- forAll genTestData
+
+    clevelandProp $ do
+      setup@Setup{seller, buyer, moneyFA2} <- testSetup testData
+      contract <- originateMarketplaceContract setup
+      consumer <- originateSimple "balance-response-consumer" [] (contractConsumer @[FA2.BalanceResponseItem])
+
+      withSender seller $
+        sell testData setup contract
+      withSender buyer $
+        buy setup contract
+
+      let expectedFee = ceiling @Double @Natural (fromIntegral (testSalePrice * testFeePercent) / 100)
+      let expectedSellerBalance = testSalePrice - expectedFee
+      sellerBalance <- balanceOf moneyFA2 consumer moneyTokenId seller
+      sellerBalance @== expectedSellerBalance
+
+hprop_NFT_is_transferred_to_buyer :: Property
+hprop_NFT_is_transferred_to_buyer  =
+  property $ do
+    testData <- forAll genTestData
+    clevelandProp $ do
+      setup@Setup{seller, buyer, nftFA2} <- testSetup testData
+      contract <- originateMarketplaceContract setup
+      consumer <- originateSimple "balance-response-consumer" [] (contractConsumer @[FA2.BalanceResponseItem])
+
+      withSender seller $
+        sell testData setup contract
+      withSender buyer $
+        buy setup contract
+
+      sellerNftBalance <- balanceOf nftFA2 consumer nftTokenId seller
+      sellerNftBalance @== 0
+      buyerNftBalance <- balanceOf nftFA2 consumer nftTokenId buyer
+      buyerNftBalance @== 1
+
+hprop_Contracts_balance_is_zero_after_a_sale_is_concluded :: Property
+hprop_Contracts_balance_is_zero_after_a_sale_is_concluded =
+  property $ do
+    testData <- forAll genTestData
+
+    clevelandProp $ do
+      setup@Setup{seller, buyer, nftFA2, moneyFA2} <- testSetup testData
+      contract <- originateMarketplaceContract setup
+      consumer <- originateSimple "balance-response-consumer" [] (contractConsumer @[FA2.BalanceResponseItem])
+
+      withSender seller $
+        sell testData setup contract
+      withSender buyer $
+        buy setup contract
+
+      contractTokenBalance <- balanceOf moneyFA2 consumer moneyTokenId contract
+      contractNftBalance <- balanceOf nftFA2 consumer nftTokenId contract
+      contractTezBalance <- getBalance contract
+      contractTokenBalance @== 0
+      contractNftBalance @== 0
+      contractTezBalance @== 0
```

Comment:
Good idea, I agree that this use case comes up often enough to justify expanding cleveland's public API a bit more and add `@@==` and `@@/=`.

I would also suggest that, since we're doing this, we should remove `checkBalance` from the public API, as it's essentially a composition of `getBalance` and  `@@==`.

Anyway, let's continue this conversation over in this morley ticket: https://gitlab.com/morley-framework/morley/-/issues/592